internal/pkg/cli/deploy/env.go

```diff
@@ -157,6 +159,7 @@ func NewEnvDeployer(in *NewEnvDeployerInput) (*envDeployer, error) {
 
 		appCFN:      deploycfn.New(defaultSession, deploycfn.WithProgressTracker(os.Stderr)),
 		envDeployer: cfnClient,
+		tmplGetter:  cfnClient,
```

Comment:
Can we not use `envDeployer: cfnClient` instead of creating new here?

Reply:
My intuition is that the template getter should not necessarily = env deployer, because their purposes are different. In this case, they just happen to be the same instance under the hood, partially because of how our codebase is currently structure. 

A counter argument would be that "it makes sense if getting a deployed template is also a responsibility of a deployer". If we prefer this line of thinking, I'm fine as well. By default, I try to separate responsibilities as fine-grind as possible, under the constraint of time & effort

Reply:
I think agree with the counter argument, also it is better to have separate responsibilities as fine-grind as possible. Let's keep as is.